DQM/BeamMonitor/plugins/AlcaBeamMonitor.cc

```diff
@@ -76,41 +78,24 @@ AlcaBeamMonitor::AlcaBeamMonitor(const ParameterSet& ps)
   for (vector<string>::iterator itV = varNamesV_.begin(); itV != varNamesV_.end(); itV++) {
     for (multimap<string, string>::iterator itM = histoByCategoryNames_.begin(); itM != histoByCategoryNames_.end();
          itM++) {
-      if (itM->first == "run") {
-        histosMap_[*itV][itM->first][itM->second] = nullptr;
-      } else {
-        positionsMap_[*itV][itM->first][itM->second] = 3 * numberOfValuesToSave_;  //value, error, ok
-        ++numberOfValuesToSave_;
-      }
+      histosMap_[*itV][itM->first][itM->second] = nullptr;
     }
   }
-
-  //  beamSpotsMap_["BF"] = map<LuminosityBlockNumber_t,BeamSpot>();//For each lumi the beamfitter will have a result
-  //  beamSpotsMap_["PV"] = map<LuminosityBlockNumber_t,BeamSpot>();//For each lumi the PVfitter will have a result
-  //  beamSpotsMap_["DB"] = map<LuminosityBlockNumber_t,BeamSpot>();//For each lumi we take the values that are stored in the database, already collapsed then
-  //  beamSpotsMap_["SC"] = map<LuminosityBlockNumber_t,BeamSpot>();//For each lumi we take the beamspot value in the file that is the same as the scaler for the alca reco stream
-  //  beamSpotsMap_["BF"] = 0;//For each lumi the beamfitter will have a result
-  //  beamSpotsMap_["PV"] = 0;//For each lumi the PVfitter will have a result
-  //  beamSpotsMap_["DB"] = 0;//For each lumi we take the values that are stored in the database, already collapsed then
-  //  beamSpotsMap_["SC"] = 0;//For each lumi we take the beamspot value in the file that is the same as the scaler for the alca reco stream
 }
 
 AlcaBeamMonitor::~AlcaBeamMonitor() {
-  if (theBeamFitter_ != nullptr) {
-    delete theBeamFitter_;
-  }
-
-  if (thePVFitter_ != nullptr) {
-    delete thePVFitter_;
-  }
+  delete theBeamFitter_;
```

Comment:
How about changing these two member data to be `std::unique_ptr<>`? Then no delete is needed since it will be handled automatically.

Reply:
sure, implementing it now

Reply:
I have also removed the denstructor at this point as it was an empty function